### PR TITLE
Add support for Field, Source and Locked parameters to AccountUpdate()

### PR DESCRIPTION
**Pull request by buckket**
A few parameters were missing for the `update_credentials` endpoint.

Reference: https://docs.joinmastodon.org/api/rest/accounts/#patch-api-v1-accounts-update-credentials

### CODE REVIEW

File: accounts.go [hunk 1]
```diff
@@ -38,6 +39,15 @@ type Field struct {
 	VerifiedAt time.Time `json:"verified_at"`
 }
 
+// Source is a Mastodon account profile field
+type Source struct {
```
Comment (mattn) [inline accounts.go]: Is this Source possible to be renamed to AccountSource?

**Comment by mattn** [comment]
Also could you please modify test? (I think modifying small is enoguh)

File: accounts.go [hunk 2]
```diff
@@ -79,6 +92,26 @@ func (c *Client) AccountUpdate(ctx context.Context, profile *Profile) (*Account,
 	if profile.Note != nil {
 		params.Set("note", *profile.Note)
 	}
+	if profile.Locked != nil {
+		params.Set("locked", strconv.FormatBool(*profile.Locked))
+	}
+	if profile.Fields != nil {
+		for idx, field := range *profile.Fields {
```
Comment (buckket) [inline accounts.go]: Mastodon explicitly states only 4 entries are allowed, but I figured it’s best letting the server complain.

File: accounts.go (hunk 1 above)
Comment (buckket) [reply accounts.go]: Done.

**Comment by mattn** [comment]
Thank you

**Pull request merged**